airflow/api_fastapi/parameters.py

```diff
@@ -144,17 +146,24 @@ def depends(self, dag_display_name_pattern: str | None = None) -> _DagDisplayNam
         return self.set_value(dag_display_name_pattern)
 
 
+# SortParam Implementations
 class SortParam(BaseParam[str]):
     """Order result by the attribute."""
 
-    attr_mapping = {
-        "last_run_state": DagRun.state,
-        "last_run_start_date": DagRun.start_date,
-    }
-
-    def __init__(self, allowed_attrs: list[str]) -> None:
+    def __init__(
+        self,
+        allowed_attrs: list[str],
+        attr_mapping: dict[str, Any],
+        model: type[BaseModel],
+        order_column: Column,
+        to_replace: dict[str, str] | None,
+    ) -> None:
         super().__init__()
         self.allowed_attrs = allowed_attrs
+        self.attr_mapping = attr_mapping
+        self.model = model
```

Comment:
I'm not sure why you need an extra `model`. Can't we use the `attr_mapping` ? That already gives the appropriate model+column ?

Reply:
Since your last suggestion included the model in the Annotated definition, I haven't deleted the model part.